pkg/controllers/build/application_controller.go

```diff
@@ -18,36 +18,210 @@ package build
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	buildv1alpha1 "github.com/projectriff/system/pkg/apis/build/v1alpha1"
+	knbuildv1alpha1 "github.com/projectriff/system/pkg/apis/thirdparty/knative/build/v1alpha1"
+	"github.com/projectriff/system/pkg/controllers/build/resources"
+	"github.com/projectriff/system/pkg/controllers/build/resources/names"
+	"github.com/projectriff/system/pkg/tracker"
 )
 
 // ApplicationReconciler reconciles a Application object
 type ApplicationReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+	Tracker tracker.Tracker
 }
 
 // +kubebuilder:rbac:groups=build.projectriff.io,resources=applications,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=build.projectriff.io,resources=applications/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=build.knative.dev,resources=builds,verbs=get;list;watch
```

Comment:
You're doing far more than just reading the build resource

```suggestion
// +kubebuilder:rbac:groups=build.knative.dev,resources=builds,verbs= get;list;watch;create;update;patch;delete
```